### ISSUE TITLE
TF 0.12 - TypeMap validation function gets placeholder values from variables

### ISSUE THREAD

**Issue by alexsomesan** [hashicorp/terraform#21419]
### Terraform Version
```
Terraform v0.12.0
+ provider.kubernetes v1.7.0
```

### Terraform Configuration Files

```hcl
variable "quota_pods" {
  default = 5
}
variable "quota_memory_request" {
  default = "4Gi"
}

resource "kubernetes_resource_quota" "resource" {
  metadata {
    name = "quota"
  }

  spec {
    hard = {
      pods              = var.quota_pods
      "requests.memory" = var.quota_memory_request

      #   pods = 5
      #   "requests.memory" = "4Gi"
    }
  }
}
```

### Debug Output

https://gist.github.com/alexsomesan/2fe40bfee0f0a07d2d036ba1a2ee6f52

### Crash Output

N/A

### Expected Behavior

Planning should execute successfully and display the plan output.

### Actual Behavior

Validation errors are shown that indicate the contents of the values in the `hard` map are not the ones passed in through the corresponding variables.
Further digging reveals that all the map's values are in fact the same UUID string.

### Steps to Reproduce

1. `terraform init`
2. `terraform plan`

### Additional Context

When the same values are passed in as literals directly in the map block, the plan executes as expected and the plan output displays the correct values.
To reproduce, uncomment the two commented lines containing literals and comment out the existing ones.

### References

https://github.com/terraform-providers/terraform-provider-kubernetes/issues/452

**Comment by alexsomesan** [comment]
This particular case is not fixed by #21443.
Below is the behaviour with a Terraform binary built from the current tip of master.

```
~/workspace/terraform-provider-kubernetes/repro(master*) » ~/bin/terraform version                                                                                                         alex@alexs-macbook-3
Terraform v0.12.2-dev
+ provider.kubernetes v1.7.0

------------------------------------------------------------
~/workspace/terraform-provider-kubernetes/repro(master*) » ~/bin/terraform plan                                                                                                            alex@alexs-macbook-3

Error: spec.0.hard.pods ("74D93920-ED26-11E3-AC10-0800200C9A66"): quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'

  on main.tf line 8, in resource "kubernetes_resource_quota" "resource":
   8: resource "kubernetes_resource_quota" "resource" {



Error: spec.0.hard.requests.memory ("74D93920-ED26-11E3-AC10-0800200C9A66"): quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'

  on main.tf line 8, in resource "kubernetes_resource_quota" "resource":
   8: resource "kubernetes_resource_quota" "resource" {


------------------------------------------------------------
```

**Comment by alexsomesan** [comment]
Update: the fix indeed is working as expected. I was not seeing it because I had omitted to bump the SDK and rebuild the provider (only updated terraform).
